src/evo/dmn_types.h

```diff
@@ -6,39 +6,50 @@
 #define BITCOIN_EVO_DMN_TYPES_H
 
 #include <amount.h>
+#include <serialize.h>
+
 #include <cassert>
 #include <string_view>
 
-class CDeterministicMNType
-{
-public:
-    uint8_t index;
-    int32_t voting_weight;
-    CAmount collat_amount;
-    std::string_view description;
+enum class MnType : uint16_t {
+    Regular = 0,
+    HighPerformance = 1,
 };
 
-namespace MnType {
-constexpr auto Regular = CDeterministicMNType{
-    .index = 0,
-    .voting_weight = 1,
-    .collat_amount = 1000 * COIN,
-    .description = "Regular",
-};
-constexpr auto HighPerformance = CDeterministicMNType{
-    .index = 1,
-    .voting_weight = 4,
-    .collat_amount = 4000 * COIN,
-    .description = "HighPerformance",
+template<> struct is_serializable_enum<MnType> : std::true_type {};
+
+struct CMnType
+{
+    const int32_t voting_weight;
+    const CAmount collat_amount;
+    const std::string_view description;
+
+    [[nodiscard]] static constexpr CMnType Regular() {
```

Comment:
>  - Braces on new lines for classes, functions, methods.

https://github.com/dashpay/dash/blob/master/doc/developer-notes.md#coding-style-c
```suggestion
    [[nodiscard]] static constexpr CMnType Regular()
    {
```
same below